pa4/checker-pa4.ml

```diff
@@ -30,23 +30,34 @@ let rec is_subtype t1 t2 =    (*checking if t1 is subtype of t2*)
 		with
 		| _ -> false (* x is undefined- Int, String, Bool*)
 		end
-	| _, _ -> (*TODO: check the class note like for SELF_TYPE*) false   
+	| SELF_TYPE(x), SELF_TYPE(y) -> 
+		if x = y then true
```

Comment:
You can just make this this `x = y`. Having a boolean statement in a if statement wrap a boolean is not required.